### ISSUE TITLE
Unable to create consumer with nats 2.9.10

### ISSUE THREAD

**Issue by cortopy** [nats-io/nats.rs#778]
Make sure that these boxes are checked before submitting your issue -- thank you!

 - [x] Included below version and environment information
 - [x] Included a [Minimal, Complete, and Verifiable example] (https://stackoverflow.com/help/mcve)

#### NATS version (`grep 'name = "nats"' Cargo.lock -A 1`)
server: 2.9.10
async-nats: 0.24.0 and 0.25.0
nats: 0.23.1

#### rustc version (`rustc --version` - we support Rust 1.41 and up)
1.65

#### OS/Container environment:
Debian

#### Steps or code to reproduce the issue:

My app creates a consumer using `get_or_create_consumer` (async). It used to work great until today I upgraded nats server to 2.9.10. Reverting the server back to 2.9.9 fixes the issue, so it looks like some kind of problem with newer version only.

When calling `get_or_create_consumer` , async nats fails with `data did not match any variant of untagged enum Response`.

Before the error, async_nats logs this in debug:
```
2022-12-21T21:02:35.354068Z DEBUG async_nats::jetstream::context: JetStream request sent: b"{\"config\":{\"ack_policy\":\"explicit\",\"deliver_policy\":\"all\",\"deliver_subject\":\"my-delivery-subject\",\"durable_name\":\"consumer-0\",\"filter_subject\":\"MY-STREAM.0.*\",\"replay_policy\":\"instant\"},\"stream_name\":\"MY-STREAM\"}" 
```

I've tried with async nats versions 0.24.0 and 0.25.0 with same results.
Containers I have using nats sync version 0.23.1 call `add_consumer` and this works fine in all versions of nats server for different consumers but similar configuration.

#### Expected result:
Creates or gets info from consumer

#### Actual result:
Call fails for unknown reasons

**Comment by rolyatmax** [comment]
We just ran into the same issue deploying the `nats:latest` docker image (v2.9.10). Pinning to `nats:2.9.9` fixed it for now.

cc @paulgb @pretentious7

**Comment by Jarema** [comment]
Thanks for the report.
Taking a look at this one.

**Comment by Jarema** [comment]
This was caused becase `consumer::Info`  expected `cluster` to be always present.
#779  hotfixes it deserializing it with default fields.

Next release will make it Option. I didn't want to make it Option outright, for patch release, as that is a breaking change in a client.

**Comment by Jarema** [comment]
Hotfix released in #780

**Comment by cortopy** [comment]
thanks so much @Jarema for the such a quick fix!
